admin/endpoints/class-endpoint-indexable.php

```diff
@@ -34,18 +36,26 @@ public function __construct( WPSEO_Indexable_Service $service ) {
 	 * @return void
 	 */
 	public function register() {
-		// Register fetch config.
-		register_rest_route( self::REST_NAMESPACE, self::ENDPOINT_SINGULAR, array(
-			'methods'             => 'GET',
-			'callback'            => array(
-				$this->service,
-				'get_indexable',
-			),
-			'permission_callback' => array(
-				$this,
-				'can_retrieve_data',
-			),
-		) );
+		$endpoints = array();
+
+		$endpoints[] = new WPSEO_Endpoint_Factory(
+			self::REST_NAMESPACE,
+			self::ENDPOINT_SINGULAR,
+			array( $this->service, 'get_indexable' ),
+			array( $this, 'can_retrieve_data' )
+		);
+
+		$endpoints[] = new WPSEO_Endpoint_Factory(
+			self::REST_NAMESPACE,
+			self::ENDPOINT_SINGULAR,
+			array( $this->service, 'patch_indexable' ),
+			array( $this, 'can_store_data' ),
+			'PATCH'
```

Comment:
As mentioned in another comment, this should preferably use `POST, PUT, PATCH` in parity with core. I recommend to use the predefined constant `WP_REST_Server::EDITABLE`.

Reply:
I am not sure we should use that constant because we've made a choice to not support/handle `POST` and `PUT` request.

Reply:
As far as I understood, the decision was not to support creating indexables in that regard. Per core's definition, it doesn't matter whether you make a `POST`, `PATCH` or `PUT` request to a single existing resource, it always means that resource should be edited. The three different request methods are there for maximum compatibility with clients. Only using `PATCH` for something where core endpoints would support all the above three would be unexpected IMO.

Reply:
`POST`, `PATCH` AND `PUT` are different HTTP verbs. If these all do the same thing there is no reason to support them all. Right now we should only introduce `PATCH` because changing an object is the only thing we support.

We don't support adding a new object (`POST`) or replacing an object (`PUT`) so we shouldn't implement those. I think WordPress is wrong here. The verbs have well-defined meanings, and we should follow those.

Reply:
@atimmer I don't disagree, but we should consider that users of the WP REST API may find that unexpected. A reason for why core uses `POST, PUT, PATCH` might be https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/rest-api/class-wp-rest-server.php#L312